lib/Conversion/TorchToTosa/TorchToTosa.cpp

```diff
@@ -3702,28 +3702,67 @@ LogicalResult ConvertAtenOp<AtenArangeStartStepOp>::matchAndRewrite(
         op, "unimplemented: pin_memory must be either None or false");
   }
 
-  int64_t start, step, end;
-  if (!matchPattern(op.getStart(), m_TorchConstantInt(&start)))
+  auto matchIntOrDouble =
```

Comment:
Wondering if [convertScalarToDtype](https://github.com/Xilinx/torch-mlir/blob/d6f1d9b0a1250f039360b15509783b8656bf8a79/lib/Conversion/Utils/Utils.cpp#L251) could have been used, as [TorchToLinalg](https://github.com/Xilinx/torch-mlir/blob/d6f1d9b0a1250f039360b15509783b8656bf8a79/lib/Conversion/TorchToLinalg/TensorConstructors.cpp#L290) uses, and checked if the result is constant folded to a constant of the appropriate type?

Perhaps this would be too complex, but I'm slightly concerned about using the C++ compiler's conversions rather than the e.g. the `arith` dialect to specify the conversions.

Reply:
I'll try to do that, and update this comment with the result.

Reply:
After some prototyping, I am leaning more towards not relying on `convertScalarToDtype` as it generates some `unrealized_conversion_cast` between `torch.int` and `integer` types, and `arith/math` operations. Also, I wonder if creating `arith/math` operations wouldn't be a `antipattern` (for the lack of a better word) of lowering one dialect at a time (?)

Reply:
Okay, and thanks for checking if it would work out.

It isn't really an anti-pattern as it is using those dialects only as a calculator as I understand you require a constant of a given type in the end.

I also understand from the call that you have also checked the validity of the using the C++ conversions for the conversions you are implementing and are convinced they are appropriate (as opposed to earlier when you thought they could be different to what `arange` requires).

Reply:
Yes, everything is sorted out. There is no problem with conversions now. It was just a wrong intuition.